rust/impl/ca/ca/context/programstate.rs

```diff
@@ -0,0 +1,64 @@
+
+pub( crate ) mod private
+{
+  /// Execution statement of a program
+  #[ derive( Debug, Default ) ]
+  pub struct ProgramState
+  {
+    /// Current instruction number
+    current_pos : usize
+  }
+
+  impl ProgramState
+  {
+    /// Sets current instruction
+    pub fn set_pos( &mut self, pos : usize )
+    {
+      self.current_pos = pos;
+    }
+
+    /// Returns the number of next instruction
+    pub fn next( &self ) -> usize
```

Comment:
I think it is the extra methods. It is enough to get current position and decrement/increment it. 

What do you think?

Reply:
I think it's more clearer in user code.

The most commonly used cases are:
- set point to previous command to be able go to current instruction later(with execution current instruction)
- set point to current command - will go to the next command(skip execution current instruction)
Controversial option:
- skip next command - go_next do it in place, but user may want to do it later. 
> E.g. `.loop .preset .do_something` where `.loop` will create a point after next instruction, '.preset` executs once and can be used anywhere else(do not create a point), and another comands

Less reason has another variants

Thats why next, I think it's ok